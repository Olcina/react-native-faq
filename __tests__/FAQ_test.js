import React from 'react';
import { FAQ } from '../index';

import renderer from 'react-test-renderer';

const action_example = () => {
    console.log("I'm an action!");

}

const bullets = [
    "I'm a bullet",
    "Second Bullet"
]

const question = {
    question: "Did I render?",
    reply: "It seems so:",
    bullets: bullets,
    actionText: "I'm ready to help!",
    onClick: action_example,
}
// render two 
const questions = [question,question]

test('FAQ renders correctly', () => {
    const tree = renderer.create(
        <FAQ
            title="A FAQ test"
            questions={questions}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});