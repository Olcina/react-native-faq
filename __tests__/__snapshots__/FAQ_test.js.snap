// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      A FAQ test
    </Text>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "lightgrey",
            "borderRadius": 5,
            "margin": 2,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            undefined,
          ]
        }
      >
        Did I render?
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            undefined,
          ]
        }
      >
        It seems so:
      </Text>
      <View
        style={
          Object {
            "display": "flex",
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "marginLeft": "10%",
              },
              undefined,
            ]
          }
        >
          I'm a bullet
        </Text>
        <Text
          style={
            Array [
              Object {
                "marginLeft": "10%",
              },
              undefined,
            ]
          }
        >
          Second Bullet
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": 5,
            "margin": 5,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 25,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          I'm ready to help!
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "lightgrey",
            "borderRadius": 5,
            "margin": 2,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            undefined,
          ]
        }
      >
        Did I render?
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            undefined,
          ]
        }
      >
        It seems so:
      </Text>
      <View
        style={
          Object {
            "display": "flex",
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "marginLeft": "10%",
              },
              undefined,
            ]
          }
        >
          I'm a bullet
        </Text>
        <Text
          style={
            Array [
              Object {
                "marginLeft": "10%",
              },
              undefined,
            ]
          }
        >
          Second Bullet
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": 5,
            "margin": 5,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 25,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          I'm ready to help!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
